order-book.js

```diff
@@ -0,0 +1,99 @@
+const reconcileOrder = (existingBook, incomingOrder) => {
+
+  switch (existingBook.length) {
+    case 0:
+      return toBook(existingBook, incomingOrder)
+
+    default:
+      return determineOrder(existingBook, incomingOrder)
+  }
+
+}
+
+const determineOrder = (existing, incoming) => {
+
+  for (let i = 0; i < existing.length; i++) {
```

Comment:
This loop should be in `reconcileOrder` allowing your sub-function to deal with a single order at a time.